src/import/importd.c

```diff
@@ -1373,6 +1374,28 @@ static bool manager_check_idle(void *userdata) {
         return hashmap_isempty(m->transfers);
 }
 
+static int xml_introspect(const char *interface_name) {
+        static const bus_object_implementation *const impls[] = {
+                &manager_object,
+                &transfer_object,
+        };
+        unsigned i;
+        int r;
+
+        for (i = 0; i < ELEMENTSOF(impls); i++)
+                if (streq(interface_name, impls[i]->interface))
+                        goto found;
```

Comment:
i'd avoid needless jumps/labels. Maybe just do `struct bus_object_implementation *found = NULL;` at the top, then assign `found` in the loop and break for it, and then do `if (!found) return log_error(…`. Much more readable I think.

Reply:
It's not needless. With the jump, the function is concise and clear. With `*found`, I need to a) do artificial initialization of the pointer and b) introduce a duplicated check for error. It's one of those cases where either `for ... else` from Python or a `goto` works much better then "structured code" or whatever it's called.